ocfweb/api/session_tracking.py

```diff
@@ -0,0 +1,111 @@
+import json
+from functools import partial
+from ipaddress import ip_address
+
+from django.conf import settings
+from django.http import HttpResponse
+from django.http import HttpResponseBadRequest
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_POST
+from ocflib.infra.ldap import ldap_ocf
+from ocflib.infra.ldap import OCF_LDAP_HOSTS
+from ocflib.infra.net import is_ocf_ip
+from ocflib.lab.stats import get_connection
+
+from ocfweb.caching import cache
+
+get_connection = partial(
+    get_connection,
+    user=settings.OCFSTATS_USER,
+    password=settings.OCFSTATS_PASSWORD,
+    db=settings.OCFSTATS_DB,
+)
+
+
+@require_POST
+@csrf_exempt
+def log_session(request):
+    """Primary API endpoint for session tracking.
+
+    Desktops have a cronjob that calls this endpoint, replacing
+    the functionality that used to be in ocf/labstats.
+    """
+
+    if not is_ocf_ip(ip_address(request.META['REMOTE_ADDR'])):
+        return HttpResponse('Not Authorized', status=401)
```

Comment:
i don't think this ip is ever going to be accurate? you'll probably see the IP of some load balancer

needs either ocfweb to use proxy protocol or [rpaf](https://github.com/yelp/wsgi-mod-rpaf) or something

Reply:
Couldn't we set a `X-Forwarded-For` header at the load balancer in nginx and then use that header here? I think we'd also need to have Apache on death forward the IP too though, so that's a bit more awkward.